File: Fruit.go
package main
import "fmt"

type Fruit struct {
	name string
	colour string
}

var longestName = ""

func newFruit(name string, colour string) Fruit {
	var fruit Fruit;
	fruit.name = name
	if (len(name) > len(longestName)) {
		longestName = name
	}
	fruit.colour = colour
	return fruit
}

func (f Fruit) getName() string {
	return f.name
}

func (f *Fruit) setColour(colour string) {
	f.colour = colour	
}

func (f Fruit) describe() string {
	return "I am a " + f.colour + " fruit called " + f.name + "."
}

func main() {
	
	for i := 10; i > 0; i-- {
		 fmt.Print(fmt.Sprintf("%d, ", i))
	}
	fmt.Println("lift off!")

	fruits := []Fruit{ newFruit("Apple", "Green"),
					   newFruit("Strawberry", "Red"),
					   newFruit("Orange", "Orange"),
					   newFruit("Banana", "Yellow"),
					   newFruit("Grape", "Green")}
	
	fruitCount := len(fruits)
	
	fmt.Println(fmt.Sprintf("There are %d fruits...", fruitCount))
	
	for _, fruit := range fruits {
		if (fruit.getName() == "Grape") {
			fruit.setColour("Purple")
		}
		fmt.Println(fruit.describe())
	}

	fmt.Println("The longest fruit name is " + longestName + "!")

}

File: Fruit.py
class Fruit:

	longestName = ""

	@classmethod
	def getLongestName(cls):
		return cls.longestName
	
	def __init__(self, theName, theColour):
		self.name = theName
		if len(theName) > len(Fruit.longestName):
			Fruit.longestName = theName
		self.colour = theColour	
	
	def getName(self):
		return self.name
		
	def setColour(self, theColour):
		self.colour = theColour	
		
	def describe(self):
		return "I am a " + self.colour + " fruit called " + self.name + "."
	
for i in range(10, 0, -1):
	print(str(i) + ", ", end="")
print ("lift off!")

fruits = [ Fruit("Apple", "Green"),
		   Fruit("Strawberry", "Red"),
		   Fruit("Orange", "Orange"),
		   Fruit("Banana", "Yellow"),
		   Fruit("Grape", "Green") ]

fruitCount = len(fruits)

print("There are " + str(fruitCount) + " fruits...")

for fruit in fruits:
	if fruit.getName() == "Grape":
		fruit.setColour("Purple")	
	print(fruit.describe())

print("The longest fruit name is " +  Fruit.getLongestName() + "!")
